### PR TITLE
Rna fixup

### PR DESCRIPTION
**Pull request by TheEimer**
RNA is now better documented and more easily runnable. There's also an option to subsample the datasets instead of always using all instances per context.

The thing that's missing right now are more context options like filtering by solvers or GC-content, but those aren't easily extractable from our data right now, so that's a separate work package all together.

### CODE REVIEW

**Comment by TheEimer** [comment]
Wow, we should resolve the flake8 issues at some point... but I'd like to fix the dist check, I just don't 100% understand what's failing there, any hints?

File: carl/envs/rna/carl_rna.py [hunk 1]
```diff
@@ -79,23 +101,17 @@ def __init__(
...
-        # Step function has a different name in this env
-        state, reward, done = self.env.execute(action)
-        if not self.hide_context:
-            state = np.concatenate((state, np.array(list(self.context.values()))))
-        self.step_counter += 1
-        return state, reward, done, {}
-
     def _update_context(self) -> None:
+        print(self.context)
+        #self.data_location = "/home/eimer/Dokumente/git/CARL/carl/envs/rna/learna/data"
```
Comment (benjamc) [inline carl/envs/rna/carl_rna.py:106]: - [ ] Remove your dir

**Review by benjamc — CHANGES_REQUESTED** [review]
Looks fine! Cleaner than before.

I think we need to write some tests for this. A test for instantiating the env would be good. Of course we could use mock ups so we don't need to install the library everytime :)

File: carl/envs/rna/carl_rna_definitions.py [hunk 2]
```diff
@@ -1,23 +1,32 @@
 import numpy as np
 from gym import spaces
 
+#FIXME: how much sense do these make? Eterna solvers are unique and I'm not sure how to get the solvers for taneda
```
[the remark this answers is not in the document]
Comment (TheEimer) [reply carl/envs/rna/carl_rna_definitions.py:4]: Nope, this is something we should clarify with Frederik, though

File: carl/context/sampling.py [hunk 3]
```diff
@@ -167,6 +167,11 @@ def sample_contexts(
                     choices = env_bounds[k][3]
                     choice = np.random.choice(choices)
                     c[k] = choice
+                elif context_feature_type == "conditional":
```
[the remark this answers is not in the document]
Comment (TheEimer) [reply carl/context/sampling.py:170]: No, I'll extend the test before merging this

**Comment by TheEimer** [comment]
> Looks fine! Cleaner than before.
> 
> I think we need to write some tests for this. A test for instantiating the env would be good. Of course we could use mock ups so we don't need to install the library everytime :)

Instantiating will be fine, all the data and stuff is only needed at reset afaik, so we can test that part :D

**Comment by TheEimer** [comment]
I removed the code leftovers and added a test for instantiating the RNA env and changing the context.
I didn't test the conditional sampling yet because there is no sampling test at all right now - we should add that when we fix the dist as well.

**Review by benjamc — APPROVED** [review]

**Pull request merged**